de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/v1_9/simulator/JSSimulatorNodeModel.java

```diff
@@ -96,28 +94,36 @@ public ValidationError validateViewValue(JSSimulatorViewValue viewContent) {
   }
 
   @Override
-  public void saveCurrentValue(NodeSettingsWO content) {}
+  public void saveCurrentValue(NodeSettingsWO content) {
+  }
 
   @Override
   public JSSimulatorViewValue getViewValue() {
-    JSSimulatorViewValue val;
+
+    JSSimulatorViewValue value = super.getViewValue();
     synchronized (getLock()) {
-      val = super.getViewValue();
 
-      if (val == null) {
-        val = createEmptyViewValue();
+      if (value.getSimulations().length == 0) {
+        copyConfigToView(value);
       }
 
-      if (val.simulations == null && port != null && port.simulations != null) {
+      if (value.getSimulations() == null) {

```

Comment:
The null check should go before checking if the simulations returned are empty (line 106). How we have it here, we could throw NPE at 106.